console/src/app/services/interceptors/org.interceptor.ts

```diff
@@ -32,3 +36,43 @@ export class OrgInterceptor<TReq = unknown, TResp = unknown> implements UnaryInt
     }
   }
 }
+
+export function NewConnectWebOrgInterceptor(orgInterceptorProvider: OrgInterceptorProvider): Interceptor {
+  return (next) => async (req) => {
+    if (!req.header.get(ORG_HEADER_KEY)) {
+      const orgId = await firstValueFrom(orgInterceptorProvider.getOrgId());
+      req.header.set(ORG_HEADER_KEY, orgId);
+    }
+
+    return next(req).catch(orgInterceptorProvider.handleError);
+  };
+}
+
+@Injectable({ providedIn: 'root' })
+export class OrgInterceptorProvider {
```

Comment:
Do you think it would be possible to use the OrgInterceptorProvider also in the old interceptor like here: https://github.com/zitadel/zitadel/blob/2559e17b6bce9c44773039ec7b694070da7f5d86/console/src/app/services/interceptors/auth.interceptor.ts#L91 ?

Reply:
overlooked... is now fixed